### PR TITLE
changed to simple and native code instead third parties libs

### PR DESCRIPTION
**Pull request by lagden**

### CODE REVIEW

File: src/utils/bsonUrlEncoding.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 /**
  * These will take a BSON object (an database result returned by the MongoDB library) and
  * encode/decode as a URL-safe string.
  */
 
-module.exports.encode = function(obj) {
-  return base64url.encode(EJSON.stringify(obj));
-};
+function encode(obj) {
+  return encodeURIComponent(Buffer.from(EJSON.stringify(obj), 'utf8').toString('base64'));
```
Comment (bradvogel) [inline src/utils/bsonUrlEncoding.js]: The library does a bit more - it escapes URL unsafe chars: https://github.com/joaquimserafim/base64-url/blob/master/index.js#L10

Comment (lagden) [reply src/utils/bsonUrlEncoding.js]: Sure, but the `encodeURIComponent` and `decodeURIComponent` make the job more safe:

```js
encodeURIComponent('/==')
// => "%2F%3D%3D"

decodeURIComponent("%2F%3D%3D")
// => "/=="
```

Now, weird side effects of `base64-url`:

```js
escape('/==')
// => "_"

unescape('_')
// => "/==="

Comment (bradvogel) [reply src/utils/bsonUrlEncoding.js]: make sense, though is there a chance that strings previously encoded by `base64url.encode` won't be correctly decoded by `decodeURIComponent`, when the user upgrades this library?

**Comment by bradvogel** [comment]
@lagden ping on this. Still interested in merging? See my question above.

**Pull request closed without merging**